src/PhpSpreadsheet/Worksheet/Table.php

```diff
@@ -0,0 +1,471 @@
+     */
+    private $columns = [];
+
+    /**
+     * Table Style.
+     *
+     * @var TableStyle
+     */
+    private $style;
+
+    /**
+     * Create a new Table.
+     *
+     * @param string $range (e.g. A1:D4)
+     *
+     * @return $this
+     */
+    public function __construct(string $range = '', ?Worksheet $worksheet = null)
+    {
+        $this->setRange($range);
+        $this->setWorksheet($worksheet);
+        $this->style = new TableStyle();
+    }
+
+    /**
+     * Get Table name.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Set Table name.
+     *
+     * @return $this
+     */
+    public function setName(string $name)
+    {
+        $name = trim($name);
+
+        if (strlen($name) == 1 && in_array($name, ['C', 'c', 'R', 'r'])) {
+            throw new PhpSpreadsheetException('The table name is invalid');
+        }
+        if (strlen($name) > 255) {
+            throw new PhpSpreadsheetException('The table name cannot be longer than 255 characters');
+        }
+        // Check for A1 or R1C1 cell reference notation
+        if (
+            preg_match(Coordinate::A1_COORDINATE_REGEX, $name) ||
+            preg_match('/^R\[?\-?[0-9]*\]?C\[?\-?[0-9]*\]?$/i', $name)
+        ) {
+            throw new PhpSpreadsheetException('The table name can\'t be the same as a cell reference');
+        }
+        if (!preg_match('/^[A-Z_\\\\]/i', $name)) {
+            throw new PhpSpreadsheetException('The table name must begin a name with a letter, an underscore character (_), or a backslash (\)');
+        }
+        if (!preg_match('/^[A-Z_\\\\][A-Z0-9\._]+$/i', $name)) {
+            throw new PhpSpreadsheetException('The table name contains invalid characters');
+        }
+
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * Get show Header Row.
+     *
+     * @return bool
+     */
+    public function getShowHeaderRow()
+    {
+        return $this->showHeaderRow;
+    }
+
+    /**
+     * Set show Header Row.
+     *
+     * @return  $this
+     */
+    public function setShowHeaderRow(bool $showHeaderRow)
+    {
+        $this->showHeaderRow = $showHeaderRow;
+
+        return $this;
+    }
+
+    /**
+     * Get show Totals Row.
+     *
+     * @return bool
+     */
+    public function getShowTotalsRow()
+    {
+        return $this->showTotalsRow;
+    }
+
+    /**
+     * Set show Totals Row.
+     *
+     * @return  $this
+     */
+    public function setShowTotalsRow(bool $showTotalsRow)
+    {
+        $this->showTotalsRow = $showTotalsRow;
+
+        return $this;
+    }
+
+    /**
+     * Get Table Range.
+     *
+     * @return string
+     */
+    public function getRange()
+    {
+        return $this->range;
+    }
+
+    /**
+     * Set Table Cell Range.
+     *
+     * @return $this
+     */
+    public function setRange(string $range): self
+    {
+        // extract coordinate
```

Comment:
Should there be an check in this method for the minimum size of the range, e.g. if a user supplies (`A1:A1`)?

Reply:
Validated ea3263650bb10d8646df91a847ac536376de6e46